### PR TITLE
feat(blade): Alert custom icon added

### LINKED ISSUE #2019: [FEATURE]: Make Alert component's leading icon configurable
Given that we've made the Toast component to have default icons + the ability to customise the icon, We should do the same for Alert component.

### PR DESCRIPTION
**Pull request by harsh9975**
## Description

Fixes #2019 

This PR adds the feature to add custom icons for Alert which was mentioned in #2019. I referred to the PR by @anuraghazra #2000 
## Changes
Added `leading` field to add custom icons.

## Additional Information
Screenshot 2024-03-03 at 6 06 11 PM
Screenshot 2024-03-03 at 6 07 01 PM

## Component Checklist

- [ ] Update Component Status Page
- [x] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

hire me

[HIRE ME](https://harish-gautam-portfolio.vercel.app/)

### CODE REVIEW

File: .changeset/slimy-toys-sort.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@razorpay/blade': major
```
Comment (saurabhdaware) [inline .changeset/slimy-toys-sort.md:2]:
```suggestion
'@razorpay/blade': minor
```

File: packages/blade/src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -69,6 +69,12 @@ type AlertProps = {
    */
   onDismiss?: () => void;
 
+  /**
+   * Can be used to render an icon
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  leading?: ComponentType<any>;
```
Comment (saurabhdaware) [inline packages/blade/src/components/Alert/Alert.tsx:76]: I think it should be called `icon` similar to how we have in Button and Link? here we don't have anything in trailing

Comment (saurabhdaware) [inline packages/blade/src/components/Alert/Alert.tsx:76]:
```suggestion
  icon?: ComponentType<IconProps>;
```

I don't remember if this is the correct type or not 🤔 Just check once in Button component

**Review by saurabhdaware — COMMENTED** [review]
I'll discuss with team once about this change from design perspective. Otherwise added some comments

File: .changeset/slimy-toys-sort.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@razorpay/blade': minor
+---
+
+Alert Custom icon added
```
Comment (anuraghazra) [inline .changeset/slimy-toys-sort.md:5]: use conventional commit. 

feat(blade): add leading prop in Alert

File: packages/blade/src/components/Alert/Alert.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Alert/Alert.tsx:76]: We can use `IconComponent` type

File: packages/blade/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -69,6 +70,12 @@ type AlertProps = {
    */
   onDismiss?: () => void;
 
+  /**
+   * Can be used to render an icon
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (anuraghazra) [inline packages/blade/src/components/Alert/Alert.tsx:76]: Why do we need this?

File: packages/blade/src/components/Alert/Alert.tsx [hunk 5]
```diff
@@ -137,7 +144,7 @@ const Alert = ({
   const isDesktop = matchedDeviceType === 'desktop';
   const isMobile = !isDesktop;
 
-  const Icon = intentIconMap[color];
+  const Icon = icon || intentIconMap[color];
```
Comment (saurabhdaware) [inline packages/blade/src/components/Alert/Alert.tsx:147]:
```suggestion
  const Icon = icon ?? intentIconMap[color];
```

**Review by saurabhdaware — APPROVED** [review]

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**